packaging/nix/flake.nix

```
@@ -2,55 +2,51 @@
   description = "umu universal game launcher";
 
   inputs = {
-    nixpkgs = {
-      type = "github";
-      owner = "NixOS";
-      repo = "nixpkgs";
-      ref = "nixpkgs-unstable";
-    };
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
   };
 
   outputs = {
     self,
     nixpkgs,
   }: let
-    umu-launcher-src = builtins.toPath "${self}/../../";
+    inherit (nixpkgs) lib;
 
-    pkgs = import nixpkgs {
-      system = "x86_64-linux";
-      overlays = [self.overlays.default];
-    };
+    # Utility function for producing consistent rename warning messages
+    rename = old: new: lib.warn "`${old}` has been renamed to `${new}`";
 
-    version = "1.1.4";
-  in {
-    overlays.default = final: prev: let
-      py = prev.python3;
-    in {
-      umu-launcher = final.callPackage ./umu-launcher.nix {
-        inherit version;
-        umu-launcher = umu-launcher-src;
-        pyth1 = py;
-      };
+    # Supported platforms & package sets
+    platforms = lib.platforms.linux;
+    supportedPkgs = lib.filterAttrs (system: _: builtins.elem system platforms) nixpkgs.legacyPackages;
 
-      umu-run = final.callPackage ./umu-run.nix {
-        inherit version;
-        package = final.umu-launcher;
+    # Use the current revision for the default version
+    version = self.dirtyShortRev or self.shortRev or self.lastModifiedDate;
+  in {
+    overlays.default = final: prev: {
+      umu-launcher = final.callPackage ./package.nix {
+        inherit (prev) umu-launcher;
       };
-
-      umu = final.callPackage ./combine.nix {
+      umu-launcher-unwrapped = final.callPackage ./unwrapped.nix {
+        inherit (prev) umu-launcher-unwrapped;
         inherit version;
-        env = final.umu-run;
-        package = final.umu-launcher;
-        truststore = true;
-        cbor2 = true;
       };
+      # Deprecated in https://github.com/Open-Wine-Components/umu-launcher/pull/345 (2025-01-27)
+      umu = rename "umu" "umu-launcher" final.umu-launcher;
+      umu-run = rename "umu-run" "umu-launcher" final.umu-launcher;
     };
 
     formatter = builtins.mapAttrs (system: pkgs: pkgs.alejandra) nixpkgs.legacyPackages;
 
-    packages.x86_64-linux = {
-      inherit (pkgs) umu;
-      default = self.packages.x86_64-linux.umu;
-    };
+    packages =
+      builtins.mapAttrs (system: pkgs: rec {
+        default = umu-launcher;
+        inherit
+          (pkgs.extend self.overlays.default)
+          umu-launcher
+          umu-launcher-unwrapped
+          ;
```

I've reverted this back to using the overlay for now.

Performance wise, this is fine. We're still using a `pkgs` instance that is likely already in memory and then just applying our extensions to it.

If we have an overlay, it feels better to make use of it so that we aren't unnecessarily duplicating code.

If the overlay was ever dropped, this would look like:
```suggestion
        umu-launcher = pkgs.callPackage ./package.nix {
          inherit umu-launcher-unwrapped;
        };
        umu-launcher-unwrapped = pkgs.callPackage ./unwrapped.nix {
          inherit version;
        };
```